src/lib/input/input-container.ts

```diff
@@ -566,19 +578,24 @@ export class MdInputContainer implements AfterViewInit, AfterContentInit, AfterC
   private _syncAriaDescribedby() {
     if (this._mdInputChild) {
       let ids: string[] = [];
-      let startHint = this._hintChildren ?
-          this._hintChildren.find(hint => hint.align === 'start') : null;
-      let endHint = this._hintChildren ?
-          this._hintChildren.find(hint => hint.align === 'end') : null;
-
-      if (startHint) {
-        ids.push(startHint.id);
-      } else if (this._hintLabel) {
-        ids.push(this._hintLabelId);
-      }
 
-      if (endHint) {
-        ids.push(endHint.id);
+      if (this._getDisplayedMessages() === 'hint') {
+        let startHint = this._hintChildren ?
+            this._hintChildren.find(hint => hint.align === 'start') : null;
+        let endHint = this._hintChildren ?
+            this._hintChildren.find(hint => hint.align === 'end') : null;
+
+        if (startHint) {
+          ids.push(startHint.id);
+        } else if (this._hintLabel) {
+          ids.push(this._hintLabelId);
+        }
+
+        if (endHint) {
+          ids.push(endHint.id);
+        }
+      } else if (this._errorChildren) {
+        ids = this._errorChildren.map(mdError => mdError.id);
```

Comment:
We could try to add to a static `aria-describedby`, but if the user has set up some binding like `[attr.aria-describedby]="myStuff"` I'm not sure there's much we can do. I'll sync with Andrew